utils/aws_api.py

```diff
@@ -0,0 +1,369 @@
+                self.custom_sqs_filter(account, sqs, queues_without_owner)
+            self.resources[account]['sqs_no_owner'] = unfiltered_queues
+
+    def map_dynamodb_resources(self):
+        for account, s in self.sessions.items():
+            dynamodb = s.client('dynamodb')
+            tables_list = dynamodb.list_tables()
+            if 'TableNames' not in tables_list:
+                continue
+            tables = tables_list['TableNames']
+            self.resources[account]['dynamodb'] = tables
+            tables_without_owner = \
+                self.get_resources_without_owner(account, tables)
+            unfiltered_tables = \
+                self.custom_dynamodb_filter(
+                    account,
+                    s,
+                    dynamodb,
+                    tables_without_owner
+                )
+            self.resources[account]['dynamodb_no_owner'] = unfiltered_tables
+
+    def map_rds_resources(self):
+        for account, s in self.sessions.items():
+            rds = s.client('rds')
+            instances_list = rds.describe_db_instances()
+            if 'DBInstances' not in instances_list:
+                continue
+            instances = [t['DBInstanceIdentifier']
+                         for t in instances_list['DBInstances']]
+            self.resources[account]['rds'] = instances
+            instances_without_owner = \
+                self.get_resources_without_owner(account, instances)
+            unfiltered_instances = \
+                self.custom_rds_filter(account, rds, instances_without_owner)
+            self.resources[account]['rds_no_owner'] = unfiltered_instances
+
+    def get_resources_without_owner(self, account, resources):
+        resources_without_owner = []
+        for r in resources:
+            if self.has_owner(account, r):
+                continue
+            resources_without_owner.append(r)
+        return resources_without_owner
+
+    def has_owner(self, account, resource):
+        has_owner = False
+        for u in self.users[account]:
+            if resource.lower().startswith(u.lower()):
+                has_owner = True
+                break
+            if '://' in resource:
+                if resource.split('/')[-1].startswith(u.lower()):
+                    has_owner = True
+                    break
+        return has_owner
+
+    def custom_s3_filter(self, account, s3, buckets):
+        type = 's3 bucket'
+        unfiltered_buckets = []
+        for b in buckets:
+            if self.resource_has_special_name(account, type, b):
+                continue
+            # ignore buckets with special tags
+            try:
+                tags = s3.get_bucket_tagging(Bucket=b)['TagSet']
+                if self.resource_has_special_tags(account, type, b, tags):
+                    continue
+                unfiltered_buckets.append(b)
+            except Exception as e:
+                if 'NoSuchTagSet' in e.message:
```

Comment:
what are you capturing here, the `s3.get_bucket_tagging(Bucket=b)['TagSet']` statement? could we avoid an exception if we do `if 'TagSet' in s3.get_bucket...`?

Reply:
or use `s3.get_bucket_tagging(Bucket=b).get('TagSet')` ? does that work?

Reply:
refactored a bit, went for the `'TagSet' in ...` approach.